lib/error_message.ex

```diff
@@ -8,14 +8,59 @@ defmodule ErrorMessage do
   @enforce_keys [:code, :message]
   defstruct [:code, :message, :details]
 
-  @type t ::
-          %ErrorMessage{code: atom, message: String.t()}
-          | %ErrorMessage{code: atom, message: String.t(), details: any}
```

Comment:
It's not possible for the ErrorMessage to be missing the details key as it is a struct